extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -288,29 +292,42 @@ public Trigger getScheduledJob(String identity) {
             this.invoker = invoker;
         }
 
-        void execute(ZonedDateTime now, ExecutorService executor) {
+        void execute(ZonedDateTime now, ExecutorService executor, Vertx vertx) {
             if (!trigger.isRunning()) {
                 return;
             }
             ZonedDateTime scheduledFireTime = trigger.evaluate(now);
             if (scheduledFireTime != null) {
-                try {
-                    executor.execute(new Runnable() {
-                        @Override
-                        public void run() {
-                            try {
-                                invoker.invoke(new SimpleScheduledExecution(now, scheduledFireTime, trigger));
-                            } catch (Throwable t) {
-                                // already logged by the StatusEmitterInvoker
+                if (invoker.isBlocking()) {
+                    try {
+                        executor.execute(new Runnable() {
```

Comment:
Do we want to switch to an executeBlocking and use a duplicated context? I don't think it would bring anything, except that it would make Context local accessible. That could be interesting for @radcortez.

Reply:
I'd like to keep the default blocking thread pool for now.